doc/rst/technotes/ioSerializers.rst

```diff
@@ -0,0 +1,375 @@
+to the versions of ``read`` and ``readln`` that accept a ``type`` argument.
+Note that fully-instantiated generic types are still allowed.
+
+Serializer API
+--------------
+
+A Serializer must implement the ``serializeValue`` method:
+
+.. code-block:: chapel
+
+   proc Serializer.serializeValue(writer: fileWriter, const val: ?) throws
+
+The ``serializeValue`` method returns nothing, and once invoked has complete
+control over how the provided value is serialized. The given ``fileWriter`` is
+guaranteed to have a ``serializerType`` identical to the type whose
+``serializeValue`` method was called. The ``fileWriter`` is also defined to be
+non-locking.
+
+By convention Serializers will invoke a ``serialize`` method on records and
+classes, but notably may choose not to do so if the class instance is ``nil``.
+
+The 'serialize' Method
+^^^^^^^^^^^^^^^^^^^^^^
+
+The ``serialize`` method has the following signature, whose API includes the
+named arguments "writer" and "serializer":
+
+.. code-block:: chapel
+
+   proc T.serialize(writer: fileWriter(?),
+                    ref serializer: writer.serializerType) throws
+
+The ``writer`` and ``serializer`` are passed separately to help distinguish the
+method signature from other possible implementations named "serialize", as well
+as to make it slightly more convenient to call methods on the Serializer. A
+future release will standardize other methods on a Serializer that provide ways
+to serialize into common types, like lists or maps.
+
+It is an error for ``writer.serializer`` to refer to a different Serializer
+instance than the ``serializer`` argument. The Serializer is responsible for
+either passing itself to the 'serializer' argument, or if applicable can create
+a new instance of itself to pass. The appropriate choice here depends on the
+degree to which the Serializer relies on internal state, and how that internal
+state must be managed. If a copy must be made, then the ``withSerializer``
+method may be used to provide an alias.
+
+.. note::
+
+   The set of standard builtin types (e.g. ranges and domains) on which this
+   method may be invoked is currently unstable.
+
+Deserializer API
+----------------
+
+A Deserializer must implement the following methods, corresponding to the
+versions of ``fileReader.read`` that accept either a type or a value:
+
+.. code-block:: chapel
+
+   proc Deserializer.deserializeType(reader: fileReader,
+                                     type readType) : readType throws
+
+   proc Deserializer.deserializeValue(reader: fileReader,
+                                      ref val: ?readType) : void throws
+
+The ``deserializeType`` method is responsible for creating a new instance of
+the given type, and returning that new instance. By convention
+``deserializeType`` will invoke an initializer by passing in the ``reader``
+and a Deserializer. If a suitable initializer is not available, this method
```

Comment:
It may not be clear that this is talking about the deserializing-initializer. I'd update to:
"... will invoke a deserializing initializer (which is typically generated by the compiler), by passing th the ..."